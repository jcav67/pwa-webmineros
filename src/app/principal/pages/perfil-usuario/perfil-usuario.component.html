<div style="margin-top: 60px">
  <div class="w3-container">
    <form [formGroup]="frmperfilUsuario">
      <div class="row justify-content-md-center">
        <div class="col-md-4">
          <label> Nombre</label>
          <input
            formControlName="strNombre"
            class="w3-input w3-border"
            id="Nombre"
            type="text"
            placeholder="Ingrese su nombre"
          />
          <br />
          <label> Apellido</label>
          <input
            formControlName="strApellido"
            class="w3-input w3-border"
            type="text"
            placeholder="Ingrese su apellido"
          />
          <br />
          <label>Teléfono</label>
          <input
            formControlName="strTelefono"
            class="w3-input w3-border"
            type="text"
            placeholder="Ingrese su Telefono"
          />
          <br />
          <label>Tipo de documento</label>

          <select class="form-control" formControlName="strTipoDocumento">
            <option value="">-- Seleccione una opción --</option>
            <option *ngFor="let tipo of tipoDocumento" [value]="tipo.valor">
              {{ tipo.texto }}
            </option>
          </select>
          <br />
          <label> Número de documento</label>
          <input
            class="w3-input w3-border"
            formControlName="strNumeroDocumento"
            type="number"
            placeholder="Ingrese su numero de documento"
            min="1"
          />
          <label
            ><span style="color: grey; font-size: 10px">
              No use espacios o caracteres especiales</span
            ></label
          >
          <br />
          <label>Género</label>
          <select class="form-control" formControlName="strGenero">
            <option value="">-- Seleccione una opción --</option>
            <option *ngFor="let tipo of lstGenero" [value]="tipo.valor">
              {{ tipo.texto }}
            </option>
          </select>
          <br />
          <label>Tipo persona</label>
          <select class="form-control" formControlName="strTipoPersona">
            <option value="">-- Seleccione una opción --</option>
            <option *ngFor="let tipo of tipoPersona" [value]="tipo.valor">
              {{ tipo.texto }}
            </option>
          </select>
          <br />
          <button class="btn btn-success form-control" (click)="ActualizarInformacion()" >Guardar</button>
          <button class="btn btn-danger form-control mt-2">
            Eliminar cuenta
          </button>
        </div>
      </div>
    </form>
  </div>
</div>