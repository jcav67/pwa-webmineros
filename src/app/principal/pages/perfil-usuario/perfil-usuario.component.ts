import { Component, OnInit } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import Swal from 'sweetalert2';
import { PerfilUsuarioService } from '../../servicios/perfil-usuario.service';


interface TipoPersona{
  texto:string;
  valor:string;
}

interface TipoDocumento{
  texto:string;
  valor:string;
}

interface Genero{
  texto:string;
  valor:string;
}

@Component({
  selector: 'app-perfil-usuario',
  templateUrl: './perfil-usuario.component.html',
  styleUrls: ['./perfil-usuario.component.css']
})
export class PerfilUsuarioComponent implements OnInit {

  frmperfilUsuario: FormGroup = this.fb.group({
    strNombre: [, [Validators.required]],
    strApellido: [, [Validators.required]],
    strTelefono: [, [Validators.required]],
    strTipoDocumento: [, [Validators.required]],
    strNumeroDocumento: [, [Validators.required]],
    strGenero: [, [Validators.required]],
    strTipoPersona: [, [Validators.required]],
  });


  tipoDocumento:TipoDocumento[]=[
    {texto:'Cédula de ciudadanía',
    valor:'cedula de ciudadania'},

    {texto:'Tarjeta de extranjería',
    valor:'tarjeta de extranjeria'},

    {texto:'Cédula de extranjería',
    valor:'cedula de extranjeria'},

    {texto:'NIT',
    valor:'NIT'},

    {texto:'Pasaporte',
    valor:'pasaporte'},

    {texto:'Tipo de documento extranjero',
    valor:'tipo de documento extranjero'}
  ]

  lstGenero:Genero[]=[

    {texto:'Hombre',
    valor:'hombre'},
    {texto:'Mujer',
    valor:'mujer'},
    {texto:'Otro',
    valor:'otro'},
    {texto:'Prefiero no decirlo',
    valor: 'prefiero no decirlo'},

  ]

  tipoPersona:TipoPersona[]=[

    {texto:'Natural',
    valor:'natural'},
    {texto:'Jurídica',
    valor:'juridica'},

  ]

  constructor(private fb: FormBuilder, private perfilUsuario:PerfilUsuarioService) {
    this.leerDatos();
    this.llenarFormulario();
   }

  ngOnInit(): void {
    this.llenarFormulario();
  }
  ActualizarInformacion(){

    if (
      this.frmperfilUsuario.invalid ||
      this.frmperfilUsuario.get('strTipoDocumento')?.value==='' ||
      this.frmperfilUsuario.get('strTipoPersona')?.value==='' ||
      this.frmperfilUsuario.get('strGenero')?.value===''
    ) {
      Swal.fire(
        'Revisar los datos',
        'error en los datos por favor revisar los datos, todos los campos son obligatorios',
        'warning'
      );
      this.leerDatos();
      return;
    }

    this.perfilUsuario
      .ActualizarDatosMinero(
        parseInt(localStorage.getItem('idMinero')!) ,
        this.frmperfilUsuario.get('strNombre')?.value,
        this.frmperfilUsuario.get('strApellido')?.value,
        this.frmperfilUsuario.get('strTelefono')?.value,
        this.frmperfilUsuario.get('strTipoDocumento')?.value,
        this.frmperfilUsuario.get('strNumeroDocumento')?.value,
        this.frmperfilUsuario.get('strGenero')?.value,
        this.frmperfilUsuario.get('strTipoPersona')?.value

      )
      .subscribe((resp) => {
        Swal.fire('Guardado exitoso', 'Registro guardado con éxito', 'success');
        this.leerDatos();
        
      });

  }
  llenarFormulario(){
    this.frmperfilUsuario.reset({
      strNombre: localStorage.getItem('perfilNombre') || '',
      strApellido: localStorage.getItem('perfilApellido') || '',
      strTelefono: localStorage.getItem('perfilTelefono') || '',
      strTipoDocumento: localStorage.getItem('perfilTipoDoc') || '',
      strNumeroDocumento: localStorage.getItem('perfilNumDoc') || '',
      strGenero: localStorage.getItem('perfilGenero') || '',
      strTipoPersona: localStorage.getItem('perfilTipoPersona') || '',
  
  
    })
  }

  leerDatos(){
    this.perfilUsuario.leerDatosMinero(34).subscribe((resp)=>{
      for (let elem of resp){
        localStorage.setItem('perfilNombre',elem.strNombre)
      localStorage.setItem('perfilApellido', elem.strApellido)
      localStorage.setItem('perfilTelefono', elem.strTelefono)
      localStorage.setItem('perfilTipoDoc', elem.strTipoIdentificacion)
      localStorage.setItem('perfilNumDoc', elem.strIdentificacion)
      localStorage.setItem('perfilGenero', elem.strGenero)
      localStorage.setItem('perfilTipoPersona', elem.strTipoPersona)
      }
    })
  }
}
