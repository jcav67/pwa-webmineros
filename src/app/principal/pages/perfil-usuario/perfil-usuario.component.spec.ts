import { ComponentFixture, TestBed } from '@angular/core/testing';
import { PerfilUsuarioService } from '../../servicios/perfil-usuario.service';
import { PerfilUsuarioComponent } from './perfil-usuario.component';
import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing'
import { FormBuilder, FormControl, FormGroup, FormsModule, ReactiveFormsModule, Validators } from '@angular/forms';


describe('PerfilUsuarioComponent', () => {
  let component: PerfilUsuarioComponent;
  let fixture: ComponentFixture<PerfilUsuarioComponent>;
  let compile:HTMLElement;
  let service:PerfilUsuarioService;
  let httpMock:HttpTestingController;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ PerfilUsuarioComponent ],
      imports:[HttpClientTestingModule,
        ReactiveFormsModule,
        FormsModule,],
      providers:[PerfilUsuarioService, FormBuilder,
       ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(PerfilUsuarioComponent);
    component = fixture.componentInstance;
    service=TestBed.inject(PerfilUsuarioService);
    httpMock=TestBed.inject(HttpTestingController);

    fixture.detectChanges();
    compile=fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('Debe ser igual al snapshot',()=>{
    expect(compile.innerHTML).toMatchSnapshot();

  });

  test('la informacion debe guardarse en local storage',()=>{
    const request=httpMock.expectOne('http://localhost:3000/api/querys/minero/datosminero?minero=5');
    expect(request.request.method).toBe('GET');


  });

  test('Guardado de datos en el local storage',()=>{
    const dummy={
      "ok": true,
      "DatosMinero": [
          {
              "strIdentificacion": "1017222708",
              "strTipoIdentificacion": "Pasaporte",
              "strTipoPersona": "juridica",
              "strGenero": "Masculino",
              "strNombre": "Camilo Arango",
              "strApellido": "Apellido prueba1",
              "strTelefono": "3006885478"
          }
      ]
  }
    const request=httpMock.expectOne('http://localhost:3000/api/querys/minero/datosminero?minero=5');
    request.flush(dummy)
    const Nombre = localStorage.getItem('perfilNombre');
    expect(Nombre).toBe(dummy.DatosMinero[0].strNombre)
  })


});
