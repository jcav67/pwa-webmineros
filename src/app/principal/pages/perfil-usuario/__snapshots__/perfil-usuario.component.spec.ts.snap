// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerfilUsuarioComponent Debe ser igual al snapshot 1`] = `
"<div style=\\"margin-top: 60px;\\"><div class=\\"w3-container\\"><form novalidate=\\"\\" ng-reflect-form=\\"[object Object]\\" class=\\"ng-untouched ng-pristine ng-invalid\\"><div class=\\"row justify-content-md-center\\"><div class=\\"col-md-4\\"><label> Nombre</label><input formcontrolname=\\"strNombre\\" id=\\"Nombre\\" type=\\"text\\" placeholder=\\"Ingrese su nombre\\" class=\\"w3-input w3-border ng-untouched ng-pristine ng-invalid\\" ng-reflect-name=\\"strNombre\\"><br><label> Apellido</label><input formcontrolname=\\"strApellido\\" type=\\"text\\" placeholder=\\"Ingrese su apellido\\" class=\\"w3-input w3-border ng-untouched ng-pristine ng-invalid\\" ng-reflect-name=\\"strApellido\\"><br><label>Teléfono</label><input formcontrolname=\\"strTelefono\\" type=\\"text\\" placeholder=\\"Ingrese su Telefono\\" class=\\"w3-input w3-border ng-untouched ng-pristine ng-invalid\\" ng-reflect-name=\\"strTelefono\\"><br><label>Tipo de documento</label><select formcontrolname=\\"strTipoDocumento\\" class=\\"form-control ng-untouched ng-pristine ng-invalid\\" ng-reflect-name=\\"strTipoDocumento\\"><option value=\\"\\" ng-reflect-value=\\"\\">-- Seleccione una opción --</option><option value=\\"cedula de ciudadania\\" ng-reflect-value=\\"cedula de ciudadania\\"> Cédula de ciudadanía </option><option value=\\"tarjeta de extranjeria\\" ng-reflect-value=\\"tarjeta de extranjeria\\"> Tarjeta de extranjería </option><option value=\\"cedula de extranjeria\\" ng-reflect-value=\\"cedula de extranjeria\\"> Cédula de extranjería </option><option value=\\"NIT\\" ng-reflect-value=\\"NIT\\"> NIT </option><option value=\\"pasaporte\\" ng-reflect-value=\\"pasaporte\\"> Pasaporte </option><option value=\\"tipo de documento extranjero\\" ng-reflect-value=\\"tipo de documento extranjero\\"> Tipo de documento extranjero </option><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object],[object Object\\"
}--></select><br><label> Número de documento</label><input formcontrolname=\\"strNumeroDocumento\\" type=\\"number\\" placeholder=\\"Ingrese su numero de documento\\" min=\\"1\\" class=\\"w3-input w3-border ng-untouched ng-pristine ng-invalid\\" ng-reflect-name=\\"strNumeroDocumento\\"><label><span style=\\"color: grey; font-size: 10px;\\"> No use espacios o caracteres especiales</span></label><br><label>Género</label><select formcontrolname=\\"strGenero\\" class=\\"form-control ng-untouched ng-pristine ng-invalid\\" ng-reflect-name=\\"strGenero\\"><option value=\\"\\" ng-reflect-value=\\"\\">-- Seleccione una opción --</option><option value=\\"hombre\\" ng-reflect-value=\\"hombre\\"> Hombre </option><option value=\\"mujer\\" ng-reflect-value=\\"mujer\\"> Mujer </option><option value=\\"otro\\" ng-reflect-value=\\"otro\\"> Otro </option><option value=\\"prefiero no decirlo\\" ng-reflect-value=\\"prefiero no decirlo\\"> Prefiero no decirlo </option><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object],[object Object\\"
}--></select><br><label>Tipo persona</label><select formcontrolname=\\"strTipoPersona\\" class=\\"form-control ng-untouched ng-pristine ng-invalid\\" ng-reflect-name=\\"strTipoPersona\\"><option value=\\"\\" ng-reflect-value=\\"\\">-- Seleccione una opción --</option><option value=\\"natural\\" ng-reflect-value=\\"natural\\"> Natural </option><option value=\\"juridica\\" ng-reflect-value=\\"juridica\\"> Jurídica </option><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object],[object Object\\"
}--></select><br><button class=\\"btn btn-success form-control\\">Guardar</button><button class=\\"btn btn-danger form-control mt-2\\"> Eliminar cuenta </button></div></div></form></div></div>"
`;
